test failure

I am getting a few test failures on master with python 2.7 in the testAll.py suite. The output
of the `python testAll.py` command is:

```
..E......F...EE.....F..E..F.................E.EFF.E..E......................................................
======================================================================
ERROR: testDeclare (testCmd.CmdTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testCmd.py", line 251, in testDeclare
    self.assertEqual(cmd.run(), 0)
  File "/usr/local/share/eups/python/eups/cmd.py", line 240, in run
    return self.execute()
  File "/usr/local/share/eups/python/eups/cmd.py", line 167, in execute
    return ecmd.run()
  File "/usr/local/share/eups/python/eups/cmd.py", line 240, in run
    return self.execute()
  File "/usr/local/share/eups/python/eups/cmd.py", line 1324, in execute
    tag=self.opts.tag, eupsenv=myeups)
  File "/usr/local/share/eups/python/eups/app.py", line 490, in declare
    tablefile, externalFileList=externalFileList, tag=tag)
  File "/usr/local/share/eups/python/eups/Eups.py", line 2626, in declare
    self._databaseFor(eupsPathDir, dbpath).declare(product)
  File "/usr/local/share/eups/python/eups/db/Database.py", line 468, in declare
    versionFile.write(trimDir)
  File "/usr/local/share/eups/python/eups/db/VersionFile.py", line 486, in write
    if trimDir and eups.utils.commonpath([trimDir, value]) == trimDir:
  File "/usr/local/share/eups/python/eups/utils.py", line 865, in commonpath
    raise ValueError("Can't mix absolute and relative paths")
ValueError: Can't mix absolute and relative paths

======================================================================
ERROR: testRemove (testCmd.CmdTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testCmd.py", line 330, in testRemove
    eups.Eups().declare("newprod", "2.0", pdir20)
  File "/usr/local/share/eups/python/eups/Eups.py", line 2626, in declare
    self._databaseFor(eupsPathDir, dbpath).declare(product)
  File "/usr/local/share/eups/python/eups/db/Database.py", line 468, in declare
    versionFile.write(trimDir)
  File "/usr/local/share/eups/python/eups/db/VersionFile.py", line 486, in write
    if trimDir and eups.utils.commonpath([trimDir, value]) == trimDir:
  File "/usr/local/share/eups/python/eups/utils.py", line 865, in commonpath
    raise ValueError("Can't mix absolute and relative paths")
ValueError: Can't mix absolute and relative paths

======================================================================
ERROR: testUses (testCmd.CmdTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testCmd.py", line 218, in testUses
    self.assertEqual(cmd.run(), 0)
  File "/usr/local/share/eups/python/eups/cmd.py", line 240, in run
    return self.execute()
  File "/usr/local/share/eups/python/eups/cmd.py", line 167, in execute
    return ecmd.run()
  File "/usr/local/share/eups/python/eups/cmd.py", line 240, in run
    return self.execute()
  File "/usr/local/share/eups/python/eups/cmd.py", line 846, in execute
    tags=self.opts.tag, pickleFile=self.opts.pickleFile)
  File "/usr/local/share/eups/python/eups/app.py", line 307, in printUses
    eupsenv.setPreferredTags(tags)
  File "/usr/local/share/eups/python/eups/Eups.py", line 664, in setPreferredTags
    self._kindlySetPreferredTags(tags, True)
  File "/usr/local/share/eups/python/eups/Eups.py", line 699, in _kindlySetPreferredTags
    raise TagNotRecognized(str(notokay), msg="Unsupported tag(s): " + ", ".join(notokay))
TagNotRecognized: Unsupported tag(s): newest

======================================================================
ERROR: testDeclare (testDb.DatabaseTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testDb.py", line 753, in testDeclare
    self.db.declare(base2)
  File "/usr/local/share/eups/python/eups/db/Database.py", line 468, in declare
    versionFile.write(trimDir)
  File "/usr/local/share/eups/python/eups/db/VersionFile.py", line 486, in write
    if trimDir and eups.utils.commonpath([trimDir, value]) == trimDir:
  File "/usr/local/share/eups/python/eups/utils.py", line 865, in commonpath
    raise ValueError("Can't mix absolute and relative paths")
ValueError: Can't mix absolute and relative paths

======================================================================
ERROR: testDeclare (testEups.EupsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testEups.py", line 253, in testDeclare
    self.eups.declare("newprod", "1.0", pdir10, testEupsStack, table)
  File "/usr/local/share/eups/python/eups/Eups.py", line 2626, in declare
    self._databaseFor(eupsPathDir, dbpath).declare(product)
  File "/usr/local/share/eups/python/eups/db/Database.py", line 468, in declare
    versionFile.write(trimDir)
  File "/usr/local/share/eups/python/eups/db/VersionFile.py", line 486, in write
    if trimDir and eups.utils.commonpath([trimDir, value]) == trimDir:
  File "/usr/local/share/eups/python/eups/utils.py", line 865, in commonpath
    raise ValueError("Can't mix absolute and relative paths")
ValueError: Can't mix absolute and relative paths

======================================================================
ERROR: testFindProduct (testEups.EupsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testEups.py", line 158, in testFindProduct
    tag = self.eups.tags.getTag("newest")
  File "/usr/local/share/eups/python/eups/tags.py", line 123, in getTag
    raise TagNotRecognized(str(tag))
TagNotRecognized: Tag not recognized: newest

======================================================================
ERROR: testRemove (testEups.EupsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testEups.py", line 491, in testRemove
    self.eups.declare("newprod", "2.0", pdir20)
  File "/usr/local/share/eups/python/eups/Eups.py", line 2626, in declare
    self._databaseFor(eupsPathDir, dbpath).declare(product)
  File "/usr/local/share/eups/python/eups/db/Database.py", line 468, in declare
    versionFile.write(trimDir)
  File "/usr/local/share/eups/python/eups/db/VersionFile.py", line 486, in write
    if trimDir and eups.utils.commonpath([trimDir, value]) == trimDir:
  File "/usr/local/share/eups/python/eups/utils.py", line 865, in commonpath
    raise ValueError("Can't mix absolute and relative paths")
ValueError: Can't mix absolute and relative paths

======================================================================
ERROR: testDetectOutOfSync (testEups.EupsCacheTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testEups.py", line 549, in testDetectOutOfSync
    e1.declare("newprod", "1.0", pdir10, testEupsStack, table)
  File "/usr/local/share/eups/python/eups/Eups.py", line 2626, in declare
    self._databaseFor(eupsPathDir, dbpath).declare(product)
  File "/usr/local/share/eups/python/eups/db/Database.py", line 468, in declare
    versionFile.write(trimDir)
  File "/usr/local/share/eups/python/eups/db/VersionFile.py", line 486, in write
    if trimDir and eups.utils.commonpath([trimDir, value]) == trimDir:
  File "/usr/local/share/eups/python/eups/utils.py", line 865, in commonpath
    raise ValueError("Can't mix absolute and relative paths")
ValueError: Can't mix absolute and relative paths

======================================================================
FAIL: testList (testCmd.CmdTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testCmd.py", line 147, in testList
    self.assertEquals(self.out.getvalue(), outall)
AssertionError: 'base                  1.0        \tcurrent\nbase                  2.0        \ncfitsio               3006.2     \tcurrent\ndoxygen               1.5.7.1    \tcurrent\neigen                 2.0.0      \tcurrent\nmpich2                1.0.5p4    \tcurrent\npython                2.5.2      \tcurrent\npython                2.6        \ntcltk                 8.5a4      \tcurrent' != 'cfitsio               3006.2     \tcurrent\ndoxygen               1.5.7.1    \tcurrent\neigen                 2.0.0      \tcurrent\nmpich2                1.0.5p4    \tcurrent\npython                2.5.2      \tcurrent\npython                2.6        \ntcltk                 8.5a4      \tcurrent'

======================================================================
FAIL: testSetVersion (testDb.ChainFileTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testDb.py", line 405, in testSetVersion
    self.assert_(bool(flavor['declarer']))
AssertionError: False is not true

======================================================================
FAIL: testFindProductNames (testDb.DatabaseTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testDb.py", line 500, in testFindProductNames
    self.assertEquals(len(prods), 6)
AssertionError: 7 != 6

======================================================================
FAIL: testInit (testEups.EupsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testEups.py", line 83, in testInit
    self.assert_(tag in tags)
AssertionError: False is not true

======================================================================
FAIL: testList (testEups.EupsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mdevalbo/project/LSST/eups/tests/testEups.py", line 404, in testList
    self.assertEquals(prods[0].version, "2.5.2")
AssertionError: '2.6' != '2.5.2'

----------------------------------------------------------------------
Ran 108 tests in 4.329s

FAILED (failures=5, errors=8)
```

I get some of those failures with master, but not all of them.
```
.........E....E.............................F.EFF...........................................................
======================================================================
ERROR: testList (testCmd.CmdTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/timj/work/lsst/eups/tests/testCmd.py", line 163, in testList
    self.assertEqual(cmd.run(), 0)
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/cmd.py", line 240, in run
    return self.execute()
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/cmd.py", line 167, in execute
    return ecmd.run()
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/cmd.py", line 240, in run
    return self.execute()
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/cmd.py", line 509, in execute
    raw=self.opts.raw
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/app.py", line 54, in printProducts
    checkTagsList(eupsenv, tags)
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/tags.py", line 545, in checkTagsList
    msg="Unsupported tag(s): %s" % ", ".join(map(lambda t: str(t), badtags)))
TagNotRecognized: Unsupported tag(s): newest

======================================================================
ERROR: testUses (testCmd.CmdTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/timj/work/lsst/eups/tests/testCmd.py", line 218, in testUses
    self.assertEqual(cmd.run(), 0)
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/cmd.py", line 240, in run
    return self.execute()
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/cmd.py", line 167, in execute
    return ecmd.run()
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/cmd.py", line 240, in run
    return self.execute()
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/cmd.py", line 846, in execute
    tags=self.opts.tag, pickleFile=self.opts.pickleFile)
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/app.py", line 307, in printUses
    eupsenv.setPreferredTags(tags)
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/Eups.py", line 664, in setPreferredTags
    self._kindlySetPreferredTags(tags, True)
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/Eups.py", line 699, in _kindlySetPreferredTags
    raise TagNotRecognized(str(notokay), msg="Unsupported tag(s): " + ", ".join(notokay))
TagNotRecognized: Unsupported tag(s): newest

======================================================================
ERROR: testFindProduct (testEups.EupsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/timj/work/lsst/eups/tests/testEups.py", line 159, in testFindProduct
    tag = self.eups.tags.getTag("newest")
  File "/Users/timj/work/lsstsw/eups/1.5.9/python/eups/tags.py", line 123, in getTag
    raise TagNotRecognized(str(tag))
TagNotRecognized: Tag not recognized: newest

======================================================================
FAIL: testDeclare (testEups.EupsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/timj/work/lsst/eups/tests/testEups.py", line 301, in testDeclare
    self.assertEquals(self.eups.findProduct("newprod", self.eups.tags.getTag("beta")).dir, pdir10)
AssertionError: '/Users/timj/work/lsst/eups/tests/Linux/newprod/1.1' != '/Users/timj/work/lsst/eups/tests/Linux/newprod/1.0'

======================================================================
FAIL: testInit (testEups.EupsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/timj/work/lsst/eups/tests/testEups.py", line 84, in testInit
    self.assert_(tag in tags)
AssertionError: False is not true

======================================================================
FAIL: testList (testEups.EupsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/timj/work/lsst/eups/tests/testEups.py", line 405, in testList
    self.assertEquals(prods[0].version, "2.5.2")
AssertionError: '2.6' != '2.5.2'

----------------------------------------------------------------------
Ran 108 tests in 2.785s

FAILED (failures=3, errors=3)
```

--- a/python/eups/tags.py
+++ b/python/eups/tags.py
@@ -146,7 +146,7 @@ class Tags(object):
         if group is None:
             group = self.global_
 
-        if isinstance(name, basestring):
+        if isinstance(name, str):
             owner = None
         else:
             name, owner = name

--- a/tests/testApp.py
+++ b/tests/testApp.py
@@ -10,7 +10,7 @@ import sys
 import shutil
 import unittest
 import time
-from io import StringIO
+from eups.utils import StringIO
 import testCommon
 from testCommon import testEupsStack
 

--- a/tests/testCmd.py
+++ b/tests/testCmd.py
@@ -8,7 +8,7 @@ import sys
 import unittest
 import time
 import re, shutil
-import io as StringIO
+from eups.utils import StringIO
 import testCommon
 from testCommon import testEupsStack
 

--- a/tests/testDeprecated.py
+++ b/tests/testDeprecated.py
@@ -10,11 +10,11 @@ import time
 import unittest
 import testCommon
 from testCommon import testEupsStack
-from io import StringIO
+from eups.utils import StringIO
 
 # reroute the error stream defined in eups.util to newerr
 syserr = sys.stderr
-newerr = StringIO()
+newerr = StringIO.StringIO()
 sys.stderr = newerr
 import eups.utils
 reload(eups.utils)

--- a/tests/testEups.py
+++ b/tests/testEups.py
@@ -9,7 +9,7 @@ import sys
 import shutil
 import unittest
 import time
-from io import StringIO
+from eups.utils import StringIO
 import testCommon
 from testCommon import testEupsStack
 
@@ -373,7 +373,7 @@ class EupsTestCase(unittest.TestCase):
     def testDeclareStdinTable(self):
         pdir = os.path.join(testEupsStack, "Linux", "newprod")
         pdir11 = os.path.join(pdir, "1.1")
-        tableStrm = StringIO('setupRequired("python")\n')
+        tableStrm = StringIO.StringIO('setupRequired("python")\n')
         prod = self.eups.findProduct("newprod", "1.1")
         self.assert_(prod is None, "newprod is already declared")
 

